app/controllers/admin/users_controller.rb

```diff
@@ -35,6 +35,24 @@ def update
       respond_with @user, location: -> { edit_admin_user_path(@user) }
     end
 
+    def upload
+      begin
+        data = process_file_upload
+      rescue Psych::SyntaxError => e
```

Comment:
`Psych` is only used for YML files, which we aren't supporting here. So you can remove this case.